cmd/geth/main.go

```diff
@@ -400,6 +401,9 @@ func run(ctx *cli.Context) {
 
 	ethereum, err := eth.New(cfg)
 	if err != nil {
+		if datadirInUseErrRE.MatchString(err.Error()) {
+			err = fmt.Errorf("%v (another instance of geth is already running with the same data directory '%v')", err, ctx.GlobalString(utils.DataDirFlag.Name))
+		}
```

Comment:
The problem is actually that this check is only added to the `run` subcommand. When you run it with another subcommand the program will still print the old message. Try to run `geth console` twice, that should reproduce the issue.

I tested it on Windows 10 to see what happens there:
```
C:\Go\GoPath\src\github.com\ethereum\go-ethereum\cmd\geth>geth.exe
I1001 09:55:40.596052    4992 database.go:71] Alloted 16MB cache to C:\Users\bas\AppData\Roaming\Ethereum\chaindata
Fatal: blockchain db err: The process cannot access the file because it is being used by another process. 
```

The returned error differs a bit from the other supported platforms. The leveldb returns an `syscall.Errno` instance with errno 11 on linux/osx and errno 32 on Windows when the database is already opened. I would suggest checking on those error numbers instead of a string match.